File: MomentumSDK/src/commonMain/kotlin/com/mwaibanda/momentum/di/ControllerModule.kt
package com.mwaibanda.momentum.di

import com.mwaibanda.momentum.controller.AuthControllerImpl
import com.mwaibanda.momentum.controller.BillingAddressControllerImpl
import com.mwaibanda.momentum.controller.EventControllerImpl
import com.mwaibanda.momentum.controller.LocalDefaultsControllerImpl
import com.mwaibanda.momentum.controller.MealControllerImpl
import com.mwaibanda.momentum.controller.NotificationControllerImpl
import com.mwaibanda.momentum.controller.PaymentControllerImpl
import com.mwaibanda.momentum.controller.SermonControllerImpl
import com.mwaibanda.momentum.controller.TransactionControllerImpl
import com.mwaibanda.momentum.controller.UserControllerImpl
import com.mwaibanda.momentum.domain.controller.AuthController
import com.mwaibanda.momentum.domain.controller.BillingAddressController
import com.mwaibanda.momentum.domain.controller.EventController
import com.mwaibanda.momentum.domain.controller.LocalDefaultsController
import com.mwaibanda.momentum.domain.controller.MealController
import com.mwaibanda.momentum.domain.controller.MessageController
import com.mwaibanda.momentum.domain.controller.NotificationController
import com.mwaibanda.momentum.domain.controller.PaymentController
import com.mwaibanda.momentum.domain.controller.SermonController
import com.mwaibanda.momentum.domain.controller.TransactionController
import com.mwaibanda.momentum.domain.controller.UserController
import org.koin.dsl.module

val controllerModule = module {
    single<PaymentController> { PaymentControllerImpl() }
    single<TransactionController> { TransactionControllerImpl(driverFactory = get()) }
    single<UserController> { UserControllerImpl(driverFactory = get()) }
    single<BillingAddressController>{ BillingAddressControllerImpl(driverFactory = get()) }
    single<AuthController> { AuthControllerImpl(controller = get()) }
    single<LocalDefaultsController> { LocalDefaultsControllerImpl() }
    single<SermonController> { SermonControllerImpl(driverFactory = get()) }
    single<MealController> { MealControllerImpl() }
    single<NotificationController> { NotificationControllerImpl() }
    single<EventController> { EventControllerImpl() }
}

File: MomentumiOS/MomentumiOS/Di/ControllerModule.swift
//
//  ControllerModule.swift
//  MomentumiOS
//
//  Created by Mwai Banda on 11/14/22.
//  Copyright © 2022 Momentum. All rights reserved.
//

import Foundation
import TinyDi
import MomentumSDK

@TinyModule
func controllerModule(resolver: TinyDi) {
    Module(
        Single<PaymentController>(PaymentControllerImpl()),
        Single<TransactionController>(TransactionControllerImpl(driverFactory: resolver.resolve())),
        Single<LocalDefaultsController>(LocalDefaultsControllerImpl()),
        Single<UserController>(UserControllerImpl(driverFactory: resolver.resolve())),
        Single<BillingAddressController>(BillingAddressControllerImpl(driverFactory: resolver.resolve())),
        Single<AuthController>(AuthControllerImpl(controller: resolver.resolve())),
        Single<SermonController>(SermonControllerImpl(driverFactory: resolver.resolve())),
        Single<MealController>(MealControllerImpl()),
        Single<MessageController>(MessageControllerImpl())
    )
}
